apps/masters/views.py

```diff
@@ -525,3 +526,174 @@ def create(self, request, *args, **kwargs):
+                "{{Party_Old_B}}": "0.0",
+                "{{Bill_amt_wd}}": bill_amount_in_words,
+                "{{Tax_amt_wd}}": bill_amount_in_words,
+                "{{address}}": (city),
+                "{{country}}": (country),
+                "{{Doc Header}}": Doc_Header
+            }
+
+            # Generate a random filename
+            unique_code = ''.join(random.choices(string.ascii_uppercase + string.digits, k=4)) + '.pdf'
+            doc_name = 'sale order_' + unique_code
+            # Construct the full file path
+            file_path = os.path.join(settings.MEDIA_ROOT, 'doc_generater', doc_name)
+            # Ensure that the directory exists
+            os.makedirs(os.path.dirname(file_path), exist_ok=True)
+
+  #=======================================ReportLab Code Started============================          
+            elements = []
+
+            # Custom page size (11 inches wide, 10.5 inches high)
+            page_width = 11 * inch
+            page_height = 10.5 * inch
+
+            # Create the PDF document
+            doc = SimpleDocTemplate(file_path, pagesize=(page_width, page_height))
+            
+
+            # Get the default styles
+            styles = getSampleStyleSheet()
+            
+            # Modify the heading style to be bold
+            style_heading = ParagraphStyle(
+                name='Heading1',
+                parent=styles['Heading1'],
+                fontName='Helvetica-Bold',  # Set font to Helvetica-Bold to make it bold
+                fontSize=16,                # You can adjust the font size if needed
+                spaceAfter=12,              # Adjust space after heading if needed
+                alignment=1,                # Center align the text (0=left, 1=center, 2=right)
+            )
+
+           
+            # Add a bold heading
+            elements.append(Paragraph(Doc_Header, style_heading))
+
+            # Add a spacer
+            elements.append(Spacer(1, 12))
+            
+            # Add Table 1
+            table_1 = create_table_1(customer_data_for_cust_data[number_value], customer_data_for_cust_data[date_value])
+            elements.append(table_1)
+
+            table_2 = create_table_2(customer_data_for_cust_data['customer']['name'], city, country, phone, dest)
+            elements.append(table_2)
+
+            table_3 = create_table_3(product_data)
+            elements.append(table_3)
+
+            table_4 = create_table_4(total_qty, total_amt, total_txbl_amt)
+            elements.append(table_4)
+
+            table_5 = create_table_5(bill_amount_in_words, bill_amount_in_words, customer_data_for_cust_data[number_value], total_qty, total_disc_amt, round_0ff, total_txbl_amt, party_old_balance, net_value)
+            elements.append(table_5)
+
+            table_6 = create_table_6()
+            elements.append(table_6)
+
+            # Build the PDF
+            doc.build(elements)
+
```

Comment:
Rename table_1, table_2 etc... to some meaninful names

Reply:
done